Add info about possible payment refund (#590)
saleor/saleor-docs: docs/developer/checkout.mdx

}
```

:::caution
When any error will be raised during the checkout completion or the payment

> will become inactive, processed payment will be refunded or voided (in case the payment wasn't charged yet).

Reviewer: ```suggestion
will become inactive, processed payment will be refunded or voided (in case it still needs to be charged).
```

Author: I'm not sure if it's gonna be more clear. For me, it will suggest that this payment will still need to be charged.
